dapp/src/tmWallet.ts

```diff
@@ -244,12 +166,48 @@ export function createTmLockWallet(privateKey: HexString): Wallet {
 
   // Sign the transaction and send it via RPC
   async function signAndSendTransaction(txSkeleton: helpers.TransactionSkeletonType): Promise<Hash> {
-    // 1. Sign transaction
-    txSkeleton = prepareSigningEntries(txSkeleton, config.lumos, 'SECP256K1_BLAKE160');
-    txSkeleton = signTransaction(txSkeleton);
+
+    let signingAction = SigningAction.unpack(bytes.bytify(txSkeleton.signingEntries.get(0).message));
+    console.log('signingAction', JSON.stringify(signingAction, null, 4))
+
+    let dappInfo: UnpackResult<typeof DappInfo> = {
+      type: 'DappInfoV1',
+      value: {
+        name: bytes.hexify(bytes.bytifyRawString('spore')),
+        url: bytes.hexify(bytes.bytifyRawString('https://a-simple-demo.spore.pro')),
+        scriptHash: computeScriptHash({
+          codeHash: config.scripts.Spore.script.codeHash,
+          hashType: config.scripts.Spore.script.hashType,
+          args: txSkeleton.outputs.get(0).cellOutput.type!.args,
+        }),
+        schema: bytes.hexify(new Uint8Array(await fetchLocalFile('../../schemas/spore.mol'))),
+        messageType: bytes.hexify(bytes.bytifyRawString('SporeAction')),
+      }
+    };
+    let dappInfoHash = ckbHash(DappInfo.pack(dappInfo));
+    if (dappInfoHash != signingAction.message.value[0].action.infoHash) {
```

Comment:
also `console.log` the `dappInfo`.

Reply:
dappInfo is already in signingAction.

Reply:
In SignAction, there is only hash of DappInfo. On wallet side, it will fetch DappInfo from some services and then validate its hash.

Reply:
Okay, I understand something wrong here